static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -200,12 +199,10 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
     })
   }
   // course is passed and paid but no certificate yet
-  if (firstRun["status"] === "passed" && paid && !course.certificate_url) {
-    if (!exams || (exams && passedExam)) {
-      messages.push({
-        message: "You passed this course."
-      })
-    }
+  if (!course.certificate_url && course.is_passed) {
+    messages.push({
+      message: "You passed this course."
+    })
```

Comment:
I think there might be a problem with this If I'm assuming correctly.

I had a passed course (`Analog Learning 300`) in a program for which i also didn't have a certificate and i saw `Course passed` message along with a `Passed` badge this is how the dashboard looked:
<img width="858" alt="image" src="https://user-images.githubusercontent.com/34372316/223439496-9b811f5a-f6eb-453e-ad80-af3e7767085b.png">

After shifting to this PR, and adding FinalGrade record for another course i see the dashboard like below, where I don't see `Course Passed` message for `Analog Learning 300` but i see `Passed` badge where as i see `Course Passed` message for `Analog Learning 100` but no `Passed` badge. I think we need to make some tweaks in https://github.com/mitodl/micromasters/blob/master/static/js/lib/grades.js#L55 as well?

<img width="791" alt="image" src="https://user-images.githubusercontent.com/34372316/223440508-abd78149-bfa9-455b-89e2-5d124ed9aeb2.png">

Reply:
I did try reproducing your scenario, but I did see the message. So I think what is happening is that the Course Run for which you have a passing grade doen't have this flag set: `FinalGrade.objects.filter(user=self.user, status=FinalGradeStatus.COMPLETE) `

Reply:
I think I might have messed up my data locally, if you are seeing the message and the`Passed` badge with this scenario it's fine then.